tests/testthat/test-multitables.R

```diff
@@ -363,16 +374,59 @@ with_mock_crunch({
         expect_header(
             expect_POST(
                 tabBook(m, data = ds2[c("gender", "starttime")]),
-                "https://app.crunch.io/api/datasets/3/multitables/ed30c4/tabbook/"
+                "https://app.crunch.io/api/datasets/3/multitables/ed30c4/export/",
+                '{\"filter\":null,',
+                '\"weight\":"https://app.crunch.io/api/datasets/3/variables/birthyr/",',
+                '\"options\":[],"where":{"function":"select","args":[{"map":',
+                '{"66ae9881e3524f7db84970d556c34552":',
+                '{"variable":"https://app.crunch.io/api/datasets/3/variables/gender/"},',
+                '"d7c21314ca9e453c93069168681a285c"',
+                ':{"variable":"https://app.crunch.io/api/datasets/3/variables/starttime/"}}}]}'
             ),
             "Accept: application/json"
         )
     })
 
-    ## TODO: test the query shape
+    test_that("tabBook with options", {
+        expect_POST(
+            tabBook(mults[[1]],
+                    data = ds, output_format = "json", format = list(pval_colors = TRUE))
+            ,
+            "https://app.crunch.io/api/datasets/1/multitables/ed30c4/export/",
+            '{\"filter\":null,\"weight\":null,\"options\":{"format":{"pval_colors":true}}}'
+        )
+    })
+
+    test_that("tabBook warning when using format argument", {
+        expect_warning(
+            expect_POST(
+                tabBook(mults[[1]],
+                        data = ds, format = "json")
+                ,
+                "https://app.crunch.io/api/datasets/1/multitables/ed30c4/export/",
+                '{\"filter\":null,\"weight\":null,\"options\":{}}'
+            ),
+            "Use `output_format`"
+        )
+    })
+
+    test_that("tabBook warning when using legacy endpoint", {
+        with(temp.option(use.legacy.tabbook.endpoint = TRUE), {
+            expect_warning(
+                expect_POST(
+                    tabBook(mults[[1]],
+                            data = ds, output_format = "json")
+                    ,
+                    "https://app.crunch.io/api/datasets/1/multitables/ed30c4/tabbook/",
+                    '{\"filter\":null,\"weight\":null,\"options\":[]}'
+                ),
+                "The legacy tabbook endpoint has been deprecated and will be removed in the future."
+            )
+        })
+    })
 
     with_POST("https://app.crunch.io/api/datasets/1/multitables/apidocs-tabbook/", {
-        book <- tabBook(mults[[1]], data = ds, format = "json")
+        book <- tabBook(mults[[1]], data = ds, output_format = "json")
```

Comment:
shouldn’t we leave one of these intact to ensure that `format="json"` doesn’t break? (and ensure coverage for the branch of code that checks for it?)

Reply:
that was a separate test, see above

Reply:
oops, thanks